s3/src/main/scala/geotrellis/spark/io/s3/S3InputFormat.scala

```diff
@@ -77,6 +87,14 @@ abstract class S3InputFormat[K, V] extends InputFormat[K,V] with LazyLogging {
       s3client
         .listObjectsIterator(request)
         .filter(!_.getKey.endsWith("/"))
+        .filter { obj =>
+          if (extensions.isEmpty)
+            true
+          else {
+            val key = obj.getKey
+            extensions.map(key.endsWith).reduce(_ || _)
+          }
+        }
```

Comment:
No, S3InputFormat didn't have a way to filter by extension like hadoop input formats. This is addition to bring options of `S3GeoTiffRDD` and `HadoopGeoTiffRDD` to parity